fe/fe-core/src/main/java/com/starrocks/lake/snapshot/RestoreClusterSnapshotMgr.java

```diff
@@ -0,0 +1,124 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.lake.snapshot;
+
+import com.starrocks.common.Config;
+import com.starrocks.common.DdlException;
+import com.starrocks.ha.FrontendNodeType;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.server.NodeMgr;
+import com.starrocks.server.WarehouseManager;
+import com.starrocks.system.Backend;
+import com.starrocks.system.ComputeNode;
+import com.starrocks.system.Frontend;
+import com.starrocks.system.SystemInfoService;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+
+public class RestoreClusterSnapshotMgr {
+    private static final Logger LOG = LogManager.getLogger(RestoreClusterSnapshotMgr.class);
+
+    private static RestoreClusterSnapshotMgr instance;
+
+    private ClusterSnapshotConfig config;
+    private boolean oldStartWithIncompleteMeta;
+    private boolean oldResetElectionGroup;
+
+    private RestoreClusterSnapshotMgr(String clusterSnapshotYamlFile) {
+        config = ClusterSnapshotConfig.load(clusterSnapshotYamlFile);
+        // Save the old config
+        oldStartWithIncompleteMeta = Config.start_with_incomplete_meta;
+        // Allow starting with only image and no log
+        Config.start_with_incomplete_meta = true;
+        // Save the old config
+        oldResetElectionGroup = Config.bdbje_reset_election_group;
+        Config.bdbje_reset_election_group = true;
+    }
+
+    public static void init(String clusterSnapshotYamlFile, String[] args) {
+        for (String arg : args) {
+            if (arg.equalsIgnoreCase("-cluster_snapshot")) {
+                LOG.info("FE start to restore from a cluster snapshot");
+                instance = new RestoreClusterSnapshotMgr(clusterSnapshotYamlFile);
+                return;
+            }
+        }
+    }
+
+    public static boolean isRestoring() {
+        return instance != null;
+    }
+
+    public static void finishRestoring() throws DdlException {
+        RestoreClusterSnapshotMgr self = instance;
+        if (self == null) {
+            return;
+        }
+
+        try {
+            List<ClusterSnapshotConfig.Frontend> frontends = self.config.getFrontends();
+            if (frontends != null) {
+                NodeMgr nodeMgr = GlobalStateMgr.getCurrentState().getNodeMgr();
+                // Drop old frontends
+                for (Frontend frontend : nodeMgr.getOtherFrontends()) {
+                    LOG.info("Drop old frontend {}", frontend);
+                    nodeMgr.dropFrontend(frontend.getRole(), frontend.getHost(), frontend.getEditLogPort());
+                }
+
+                // Add new frontends
```

Comment:
These frontends will be lost after restart without `--cluster_snapshot`.
Adding other frontends by command `Alter system add follower xxx` is better?

Reply:
If users config the frontend list in the cluster_snapshot.yaml, it indicates the system is recovered with new machines, so the old frontends should be dropped. If users want to recover the system with old machines, they should not to config the frontend list in the cluster_snapshot.yaml.

Reply:
`If users config the frontend list in the cluster_snapshot.yaml`
The frontends added will be lost after restart, since they are not persited